app/src/main/java/org/systers/mentorship/view/fragments/EditProfileFragment.kt

```diff
@@ -102,17 +90,29 @@ class EditProfileFragment: DialogFragment() {
                     }
                 }
             }
+
             if (currentUser != editProfileBinding.user && errors.isEmpty()) {
-                profileViewModel.updateProfile(editProfileBinding.user!!)
+                updateProfile()
             }
         }
     }
 
-    override fun onDestroy() {
-        super.onDestroy()
+    private fun updateProfile() {
+        profileViewModel.updateProfile(editProfileBinding.user!!).observe(viewLifecycleOwner,
```

Comment:
Please avoid using `!! `

Reply:
I had no other choice in this case, but okay. I'll find a way to get rid of this.

Reply:
```editProfileBinding.user?.let{ profileViewModel.updateProfile(it).observe ... }```

Reply:
It's in the force-pushed commit :)